codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGenerator.kt

```diff
@@ -40,6 +41,12 @@ open class ServerServiceGenerator(
      * which assigns a symbol location to each shape.
      */
     fun render() {
+        rustCrate.lib {
+            val serviceName = codegenContext.serviceShape.id.name.toString()
```

Comment:
Is this safe to use as a rust name? i.e, if `serviceName` was a rust keyword (`mod`, `async`, etc).

Reply:
We also do this in `ServerProtocolTestGenerator` and `ServerServiceGeneratorV2`.

IIRC our symbol providers should map shapes in a way which considers reserved keywords, however we've not got a symbol provider which maps `ServiceShape`s at the moment.

https://github.com/awslabs/smithy-rs/issues/1724